test/parallel/test-repl-built-in-modules.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+const common = require('../common');
+const tmpdir = require('../common/tmpdir');
+const assert = require('assert');
+const spawn = require('child_process').spawn;
+
+tmpdir.refresh();
+
+function run(wasiPreview) {
+  let resolve;
+  const promise = new Promise((_resolve) => {
+    resolve = _resolve;
+  });
+
+  // Use -i to force node into interactive mode, despite stdout not being a TTY
+  const args = ['-i'];
+  if (wasiPreview) args.push('--experimental-wasi-unstable-preview1');
+  const child = spawn(process.execPath, args);
+
+  const input = 'require(\'events\');\nrequire(\'wasi\');';
+  let stdout = '';
+  let stderr = '';
+
+  child.stdout.setEncoding('utf8');
+  child.stdout.on('data', (c) => stdout += c);
+  child.stderr.setEncoding('utf8');
+  child.stderr.on('data', (c) => stderr += c);
+
+  child.stdin.end(input);
+
+  child.on('exit', () => {
+    resolve({ stdout, stderr });
+  });
+
+  return promise;
+}
+
+(async function() {
+  const ret1 = await run(false);
+  const ret2 = await run(true);
```

Comment:
Nit: These two lines are impossible to understand by themselves. The variable names are not useful and the boolean argument isn't either.